tests/test_config.py

```diff
@@ -171,3 +173,32 @@ def test_timeout_from_tuple():
 def test_timeout_from_config_instance():
     timeout = httpx.TimeoutConfig(timeout=5.0)
     assert httpx.TimeoutConfig(timeout) == httpx.TimeoutConfig(timeout=5.0)
+
+
+@pytest.mark.skipif(
+    not hasattr(ssl.SSLContext, "keylog_filename"),
+    reason="requires OpenSSL 1.1.1 or higher",
+)
+@pytest.mark.skipif(sys.version_info < (3, 8), reason="requires python3.8 or higher")
+def test_ssl_config_support_for_keylog_file(tmpdir):
+    old = os.getenv("SSLKEYLOGFILE")
+    if old is not None:
+        del os.environ["SSLKEYLOGFILE"]
+
+    ssl_config = httpx.SSLConfig()
+    ssl_config.load_ssl_context()
+
+    assert ssl_config.ssl_context.keylog_filename is None
+
+    filename = str(tmpdir.join("test.log"))
+    os.environ["SSLKEYLOGFILE"] = filename
+
+    ssl_config = httpx.SSLConfig()
+    ssl_config.load_ssl_context()
+
+    assert ssl_config.ssl_context.keylog_filename == filename
+
+    if old is None:
+        del os.environ["SSLKEYLOGFILE"]
+    else:
+        os.environ["SSLKEYLOGFILE"] = old
```

Comment:
Note that if the test fails we never reach this part of the code. We’d probably want to extract this logic as an override_environ() context manager helper (if pytest doesn’t provide it yet?). Not sure if we aren’t already doing this kind of override in other tests though — might be worth seeing if we can reuse any existing utility first?

Reply:
Couldn't find any override similar to this one in other tests: all other tests that set an environment variable do so without resetting the environment after the fact. Checked pytest and it uses [monkeypatch](https://docs.pytest.org/en/latest/monkeypatch.html#monkeypatching-environment-variables) for it. Another alternative is using [this](https://github.com/depop/django-concurrent-test-helper/blob/master/django_concurrent_tests/utils.py#L151).

Reply:
The monkeypatch way looks good. :)